### PR TITLE
test: add snapshot tests for help

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
tests

**Did you add tests for your changes?**
yes

**If relevant, did you update the documentation?**
NA

**Summary**
- Adds snapshot tests for help to make it easier to see the diff when something in the output changes

**Does this PR introduce a breaking change?**
no

**Other information**
- I've only added it in places where it makes sense, any other place in help we need to add this?

### CODE REVIEW

**Comment by alexander-akait** [comment]
@anshumanv Can you rebase?

**Comment by anshumanv** [comment]
@alexander-akait  done

**Comment by anshumanv** [comment]
need help resolve fail test, jest errors on obsolete snapshot

File: scripts/globalSetup.js [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+            console.log(`removed ${join(outputFolder, dir)}`);
+            rimraf.sync(join(outputFolder, dir));
+        });
+    }
+};
+
+module.exports = () => {
+    cleanupOutputDirs();
+    console.log(`\n Running tests for webpack @${version} \n`);
+};
```
Comment (alexander-akait) [inline scripts/globalSetup.js:28]: Why we need this?

File: scripts/snapshotResolver.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    // resolves from snapshot to test path
+    resolveTestPath: (snapshotFilePath, snapshotExtension) => {
+        const perVersionDir = customSnapshotDir.some((dir) => snapshotFilePath.includes(`test/${dir}`));
+        const snapshotDirName = !perVersionDir ? '__snapshots__' : `__snapshots-v${version[0]}__`;
+        return snapshotFilePath.replace(`/${snapshotDirName}`, '').slice(0, -snapshotExtension.length);
+    },
+
+    // Example test path, used for preflight consistency check of the implementation above
+    testPathForConsistencyCheck: 'some/__tests__/example.test.js',
+};
```
Comment (alexander-akait) [inline scripts/snapshotResolver.js:24]: Can be simplified, look here https://github.com/webpack-contrib/terser-webpack-plugin/blob/v4.2.3/test/helpers/snapshotResolver.js

Comment (anshumanv) [reply scripts/snapshotResolver.js:24]: great, will change

**Comment by anshumanv** [comment]
@alexander-akait what do we do about stale snapshots as for one webpack version, only half snapshots will be relevant and other half will be obsolete - failing the test

**Comment by anshumanv** [comment]
This problem is not in terset plugin repo?

**Comment by anshumanv** [comment]
Great, wip

**Comment by anshumanv** [comment]
@alexander-akait CI should be green, this resolver is good 👍

**Comment by anshumanv** [comment]
help is broken 😞 outputs all flags on --help, wip fix

**Review by alexander-akait — COMMENTED** [review]
Please use `strip-ansi` before snapshots, you can create normalizeStderr and normalizeStdout utils in our `utils` and put logic there for future improvements, tests on colors should use `.contains('<color-characters>')`

**Comment by anshumanv** [comment]
yep lets do it

File: test/help/help.test.js [hunk 3]
```diff
@@ -14,111 +11,31 @@ describe('help', () => {
...
-            expect(stdout.match(/init\|c/g)).toHaveLength(1);
-            expect(stdout.match(/loader\|l/g)).toHaveLength(1);
-            expect(stdout.match(/plugin\|p/g)).toHaveLength(1);
-            expect(stdout.match(/migrate\|m/g)).toHaveLength(1);
-            expect(stdout.match(/configtest\|t/g)).toHaveLength(1);
-            expect(stdout).toContain("To see list of all supported commands and options run 'webpack --help=verbose'.");
-            expect(stdout).toContain('CLI documentation: https://webpack.js.org/api/cli/.');
-            expect(stdout).toContain('Made with ♥ by the webpack team.');
-        }
+        expect(stripAnsi(stdout)).toMatchSnapshot();
```
Comment (jamesgeorge007) [inline test/help/help.test.js:14]: Why not making use of the `jest-serializer-ansi` snapshot serializer?
https://github.com/webpack/webpack-cli/blob/e13082221c2da01d8b8215ebc936474bf3ca1582/package.json#L73

Comment (anshumanv) [reply test/help/help.test.js:14]: yes much cleaner, lets do it

File: scripts/globalSetup.js (hunk 1 above)
Comment (alexander-akait) [inline scripts/globalSetup.js:28]: Please clarify why we need remove something?

Comment (anshumanv) [reply scripts/globalSetup.js]: no need removed

**Review by alexander-akait — APPROVED** [review]
/cc @webpack/cli-team 

Need migrate on snapshot for `version` too

**Comment by anshumanv** [comment]
> Need migrate on snapshot for version too

wip, in near future

**Review by snitin315 — APPROVED** [review]

**Review by jamesgeorge007 — APPROVED** [review]

**Comment by anshumanv** [comment]
Thanks

**Pull request merged**